internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -721,12 +722,14 @@ func buildRateLimitRule(rule egv1a1.RateLimitRule) (*ir.RateLimitRule, error) {
 				m := &ir.StringMatch{
 					Name:      header.Name,
 					SafeRegex: header.Value,
+					Invert:    header.Invert,
 				}
 				irRule.HeaderMatches = append(irRule.HeaderMatches, m)
 			case *header.Type == egv1a1.HeaderMatchDistinct && header.Value == nil:
 				m := &ir.StringMatch{
 					Name:     header.Name,
 					Distinct: true,
+					Invert:   header.Invert,
```

Comment:
invert doesnt make sense for `Distinct` . We should probably catch that and flag that in  [either in API CEL Validation](https://github.com/envoyproxy/gateway/blob/ffe029b576eef5a500058bbad3b461ac489083a3/api/v1alpha1/ratelimit_types.go#L150) or [Status](https://github.com/envoyproxy/gateway/blob/ffe029b576eef5a500058bbad3b461ac489083a3/internal/gatewayapi/backendtrafficpolicy.go#L692)

Reply:
@arkodg Added a check in Status and a corresponding test as well.

Reply:
thanks, can you also rm L736 since its NA ?

Reply:
Done!